sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/CategorizedEntity.java

```diff
@@ -3,49 +3,81 @@
 
 package com.azure.ai.textanalytics.models;
 
+import com.azure.core.annotation.Immutable;
+
 /**
  * The {@link CategorizedEntity} model.
  */
-public interface CategorizedEntity {
-    /**
-     * Get the text property: Categorized entity text as appears in the request.
-     *
-     * @return The text value.
+@Immutable
+public final class CategorizedEntity {
+    /*
+     * CategorizedEntity text as appears in the request.
+     */
+    private final String text;
+
+    /*
+     * CategorizedEntity category, such as Person/Location/Org/SSN etc
+     */
+    private final EntityCategory category;
+
+    /*
+     * CategorizedEntity sub category, such as Age/Year/TimeRange etc
+     */
+    private final String subcategory;
```

Comment:
Should subcategory also be of type enum? (I'm just not sure where the team landed, so could definitely be wrong here)

Reply:
we have decided to only maintain an enum class for category but not subcategory. Subcategory has way more enum values than category.